src/mkfs.fat.c

```diff
@@ -593,7 +594,7 @@ static void establish_params(struct device_info *info)
     }
 
     if (!hidden_sectors_by_user && info->geom_start >= 0)
-	hidden_sectors = info->geom_start;
+        hidden_sectors = htole32(info->geom_start) + part_sector;
```

Comment:
This is incorrect. `hidden_sectors` is stored in host endianity, not in little endian.

Reply:
We have now configured Travis to test dosfstools also on big endian system, but hidden sector value is not covered by any test. So this problem is not detected by automatic tests.